### ISSUE TITLE
"via" parameter in room URL gives other result when I try to join

### ISSUE THREAD

**Issue by JeanPaulLucien** [vector-im/element-web#24617]
Here is Help vampire again.

### Steps to reproduce

2 group of rooms below. Try to join to see difference.
```
https://matrix.to/#/!elPHPEplkxTcHJLIRR:feneas.org?via=feneas.org 
https://matrix.to/#/!tUpugqyPScyTwHoucv:feneas.org?via=feneas.org

https://matrix.to/#/!elPHPEplkxTcHJLIRR:feneas.org 
https://matrix.to/#/!tUpugqyPScyTwHoucv:feneas.org
``` 
**Note:** server does not exist.

#### What did you expect?
Same result for both group. Normal description why I can't join for 1st group. For 2nd group it's 404.
#### What happened instead?
Other result. Just "Failed to join" for 1st group.

### Operating system

Windows

### Application version

1.11.23

### How did you install the app?

In-app update

### Homeserver

matrix.org

### Will you send logs?

No

**Comment by JeanPaulLucien** [comment]
It's discovered case of https://github.com/vector-im/element-web/issues/24162

**Comment by t3chguy** [comment]
Duplicate of https://github.com/vector-im/element-web/issues/13247

**Comment by JeanPaulLucien** [comment]
@t3chguy in my case you know that server of room does not exist more; it seems the link without "via" provides the correct information why I can't join. You could to make the fix for my case and possible other issue will be resolved too. It could be the reference.

@t3chguy I have seen the pool request. It's totally different and will not resolve this issue.

**Comment by t3chguy** [comment]
@JeanPaulLucien the server doesn't tell us that the other server doesn't exist anymore. Just that the join failed.

**Comment by JeanPaulLucien** [comment]
@t3chguy the links from 2nd group gives the popup with normal information. I suppose that same popup should be when I try to join using links from 1st group.

**Comment by JeanPaulLucien** [comment]
2nd group:
![image](https://user-images.githubusercontent.com/24779886/220917509-84e3e5e1-edec-4db4-813f-dc2ac950211f.png)
1st group, why empty:
![image](https://user-images.githubusercontent.com/24779886/220917606-3dbf3cdd-ace9-4370-ac8d-2b2a2162121e.png)

**Comment by t3chguy** [comment]
Because the server itself gives a different error. I suggest Settings > Help & About > Submit debug logs > Download logs so you can see for yourself

**Comment by JeanPaulLucien** [comment]
1st  group. 403 when I click on the link, 502 when click Join to...
```
fetch.ts:62          GET https://matrix-client.matrix.org/_matrix/client/r0/rooms/!elPHPEplkxTcHJLIRR%3Afeneas.org/initialSync?limit=20 403
fetch @ fetch.ts:62
requestOtherUrl @ fetch.ts:264
request @ fetch.ts:205
authedRequest @ fetch.ts:156
roomInitialSync @ client.ts:7932
peek @ sync.ts:410
peekInRoom @ client.ts:6177
setupRoom @ RoomView.tsx:784
(anonymous) @ RoomView.tsx:706
componentDidMount @ RoomView.tsx:842
_s @ react-dom.production.min.js:219
Rl @ react-dom.production.min.js:259
t.unstable_runWithPriority @ scheduler.production.min.js:18
$i @ react-dom.production.min.js:122
Il @ react-dom.production.min.js:252
vl @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
$i @ react-dom.production.min.js:122
Yi @ react-dom.production.min.js:123
Wi @ react-dom.production.min.js:122
ml @ react-dom.production.min.js:237
enqueueSetState @ react-dom.production.min.js:133
f.setState @ react.production.min.js:12
setState @ MatrixChat.tsx:415
viewRoom @ MatrixChat.tsx:981
(anonymous) @ MatrixChat.tsx:698
t._invokeCallback @ Dispatcher.js:214
t.dispatch @ Dispatcher.js:189
setTimeout (async)
dispatch @ dispatcher.ts:52
showScreen @ MatrixChat.tsx:1855
(anonymous) @ routing.ts:42
p @ routing.ts:50
hashchange (async)
click @ linkify-matrix.ts:165
rageshake.ts:73 !elPHPEplkxTcHJLIRR:feneas.org is current in RVS but missing from client - clearing sticky room
e.<computed> @ rageshake.ts:73
(anonymous) @ logger.ts:44
handleRVSUpdate @ RoomListStore.ts:153
(anonymous) @ RoomListStore.ts:124
s.emit @ events.js:158
setState @ RoomViewStore.tsx:226
joinRoom @ RoomViewStore.tsx:529
onDispatch @ RoomViewStore.tsx:297
t._invokeCallback @ Dispatcher.js:214
t.dispatch @ Dispatcher.js:189
setTimeout (async)
dispatch @ dispatcher.ts:52
(anonymous) @ RoomView.tsx:1467
Promise.then (async)
(anonymous) @ RoomView.tsx:1465
ze @ react-dom.production.min.js:52
Ye @ react-dom.production.min.js:52
(anonymous) @ react-dom.production.min.js:53
Dr @ react-dom.production.min.js:100
kr @ react-dom.production.min.js:101
(anonymous) @ react-dom.production.min.js:113
Pe @ react-dom.production.min.js:292
(anonymous) @ react-dom.production.min.js:50
Mr @ react-dom.production.min.js:105
Jt @ react-dom.production.min.js:75
Zt @ react-dom.production.min.js:74
t.unstable_runWithPriority @ scheduler.production.min.js:18
$i @ react-dom.production.min.js:122
Ne @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
://vector/webapp/#/room/!elPHPEplkxTcHJLIRR:feneas.org:1 Access to fetch at 'https://matrix-client.matrix.org/_matrix/client/r0/join/!elPHPEplkxTcHJLIRR%3Afeneas.org?server_name=feneas.org' from origin 'vector://vector' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
fetch.ts:62          POST https://matrix-client.matrix.org/_matrix/client/r0/join/!elPHPEplkxTcHJLIRR%3Afeneas.org?server_name=feneas.org net::ERR_FAILED 502
fetch @ fetch.ts:62
requestOtherUrl @ fetch.ts:264
request @ fetch.ts:205
authedRequest @ fetch.ts:156
joinRoom @ client.ts:3999
await in joinRoom (async)
dis.dispatch.action @ RoomViewStore.tsx:541
i @ promise.ts:39
joinRoom @ RoomViewStore.tsx:539
onDispatch @ RoomViewStore.tsx:297
t._invokeCallback @ Dispatcher.js:214
t.dispatch @ Dispatcher.js:189
setTimeout (async)
dispatch @ dispatcher.ts:52
(anonymous) @ RoomView.tsx:1467
Promise.then (async)
(anonymous) @ RoomView.tsx:1465
ze @ react-dom.production.min.js:52
Ye @ react-dom.production.min.js:52
(anonymous) @ react-dom.production.min.js:53
Dr @ react-dom.production.min.js:100
kr @ react-dom.production.min.js:101
(anonymous) @ react-dom.production.min.js:113
Pe @ react-dom.production.min.js:292
(anonymous) @ react-dom.production.min.js:50
Mr @ react-dom.production.min.js:105
Jt @ react-dom.production.min.js:75
Zt @ react-dom.production.min.js:74
t.unstable_runWithPriority @ scheduler.production.min.js:18
$i @ react-dom.production.min.js:122
Ne @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
rageshake.ts:73 !elPHPEplkxTcHJLIRR:feneas.org is current in RVS but missing from client - clearing sticky room
```

2nd group: 403 when I click on link and 404 when I click join to
```
fetch.ts:62          GET https://matrix-client.matrix.org/_matrix/client/r0/rooms/!elPHPEplkxTcHJLIRR%3Afeneas.org/initialSync?limit=20 403
fetch @ fetch.ts:62
requestOtherUrl @ fetch.ts:264
request @ fetch.ts:205
authedRequest @ fetch.ts:156
roomInitialSync @ client.ts:7932
peek @ sync.ts:410
peekInRoom @ client.ts:6177
setupRoom @ RoomView.tsx:784
(anonymous) @ RoomView.tsx:706
componentDidMount @ RoomView.tsx:842
_s @ react-dom.production.min.js:219
Rl @ react-dom.production.min.js:259
t.unstable_runWithPriority @ scheduler.production.min.js:18
$i @ react-dom.production.min.js:122
Il @ react-dom.production.min.js:252
vl @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
$i @ react-dom.production.min.js:122
Yi @ react-dom.production.min.js:123
Wi @ react-dom.production.min.js:122
ml @ react-dom.production.min.js:237
enqueueSetState @ react-dom.production.min.js:133
f.setState @ react.production.min.js:12
setState @ MatrixChat.tsx:415
viewRoom @ MatrixChat.tsx:981
(anonymous) @ MatrixChat.tsx:698
t._invokeCallback @ Dispatcher.js:214
t.dispatch @ Dispatcher.js:189
setTimeout (async)
dispatch @ dispatcher.ts:52
showScreen @ MatrixChat.tsx:1855
(anonymous) @ routing.ts:42
p @ routing.ts:50
hashchange (async)
click @ linkify-matrix.ts:165
rageshake.ts:73 !elPHPEplkxTcHJLIRR:feneas.org is current in RVS but missing from client - clearing sticky room
e.<computed> @ rageshake.ts:73
(anonymous) @ logger.ts:44
handleRVSUpdate @ RoomListStore.ts:153
(anonymous) @ RoomListStore.ts:124
s.emit @ events.js:158
setState @ RoomViewStore.tsx:226
joinRoom @ RoomViewStore.tsx:529
onDispatch @ RoomViewStore.tsx:297
t._invokeCallback @ Dispatcher.js:214
t.dispatch @ Dispatcher.js:189
setTimeout (async)
dispatch @ dispatcher.ts:52
(anonymous) @ RoomView.tsx:1467
Promise.then (async)
(anonymous) @ RoomView.tsx:1465
ze @ react-dom.production.min.js:52
Ye @ react-dom.production.min.js:52
(anonymous) @ react-dom.production.min.js:53
Dr @ react-dom.production.min.js:100
kr @ react-dom.production.min.js:101
(anonymous) @ react-dom.production.min.js:113
Pe @ react-dom.production.min.js:292
(anonymous) @ react-dom.production.min.js:50
Mr @ react-dom.production.min.js:105
Jt @ react-dom.production.min.js:75
Zt @ react-dom.production.min.js:74
t.unstable_runWithPriority @ scheduler.production.min.js:18
$i @ react-dom.production.min.js:122
Ne @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
fetch.ts:62          POST https://matrix-client.matrix.org/_matrix/client/r0/join/!elPHPEplkxTcHJLIRR%3Afeneas.org 404
fetch @ fetch.ts:62
requestOtherUrl @ fetch.ts:264
request @ fetch.ts:205
authedRequest @ fetch.ts:156
joinRoom @ client.ts:3999
await in joinRoom (async)
dis.dispatch.action @ RoomViewStore.tsx:541
i @ promise.ts:39
joinRoom @ RoomViewStore.tsx:539
onDispatch @ RoomViewStore.tsx:297
t._invokeCallback @ Dispatcher.js:214
t.dispatch @ Dispatcher.js:189
setTimeout (async)
dispatch @ dispatcher.ts:52
(anonymous) @ RoomView.tsx:1467
Promise.then (async)
(anonymous) @ RoomView.tsx:1465
ze @ react-dom.production.min.js:52
Ye @ react-dom.production.min.js:52
(anonymous) @ react-dom.production.min.js:53
Dr @ react-dom.production.min.js:100
kr @ react-dom.production.min.js:101
(anonymous) @ react-dom.production.min.js:113
Pe @ react-dom.production.min.js:292
(anonymous) @ react-dom.production.min.js:50
Mr @ react-dom.production.min.js:105
Jt @ react-dom.production.min.js:75
Zt @ react-dom.production.min.js:74
t.unstable_runWithPriority @ scheduler.production.min.js:18
$i @ react-dom.production.min.js:122
Ne @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
rageshake.ts:73 !elPHPEplkxTcHJLIRR:feneas.org is current in RVS but missing from client - clearing sticky room
```

**Comment by t3chguy** [comment]
The request you're looking for is a `/join` request (given you clicked Join), the `/initialSync` request is done when you first navigate to the link in an attempt to preview the room if it is public & previewable.
